### PR TITLE
[Dep] Add PHP 8.1 support

### LINKED ISSUE #21: PHP 8.1 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

PHP with [version 8.1 is released](https://www.php.net/releases/8.1/index.php) and for this component the support of the new version is still missing.

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| New Feature   | yes

### CODE REVIEW

**Comment by samsonasik** [comment]
@Xerkus do you have idea to solve psalm notice:

```bash
Error: test/MiddlewareControllerTest.php:48:29: InternalMethod: Constructor Laminas\Mvc\Middleware\MiddlewareController::__construct is internal to Laminas but called from LaminasTest\Mvc\Middleware (see https://psalm.dev/175)
```

The issue seems same with that reported at https://github.com/vimeo/psalm/issues/6009

**Comment by samsonasik** [comment]
@Xerkus ok, resolved with `@psalm-suppress InternalMethod` in `MiddlewareControllerTest` 🎉

**Comment by samsonasik** [comment]
@Xerkus All green 🎉

File: test/PipeSpecTest.php [hunk 1]
```diff
@@ -33,7 +31,7 @@ public function process(
         $spec       = new PipeSpec('container_key_string', $middleware, 'another_key');
         self::assertSame(
             ['container_key_string', $middleware, 'another_key'],
-            array_values($spec->getSpec())
```
Comment (Xerkus) [inline test/PipeSpecTest.php:36]: array_values was used to make sure it is not affected by different keys by discarding keys

Comment (Xerkus) [reply test/PipeSpecTest.php:36]: Not needed anymore, implementation can't receive keys in input.

Comment (samsonasik) [reply test/PipeSpecTest.php:36]: added back with `@psalm-suppress RedundantCast` https://github.com/laminas/laminas-mvc-middleware/pull/22/commits/f30ad4c8bf0cbb348810f5570e7432e469f19c1d

**Comment by samsonasik** [comment]
@Xerkus All green 🎉

File: test/PipeSpecTest.php (hunk 1 above)
Comment (samsonasik) [reply test/PipeSpecTest.php:36]: ok, removed again https://github.com/laminas/laminas-mvc-middleware/pull/22/commits/93ff4a47d587b0681ad7e5d49b288445fb172b04

**Comment by samsonasik** [comment]
@Xerkus All green 🎉

**Pull request merged**